Library/Homebrew/test/compiler_selector_spec.rb

```diff
@@ -42,9 +42,20 @@
       expect(selector.compiler).to eq("gcc-7")
     end
 
-    it "returns gcc-6 if gcc formula offers gcc-6" do
+    it "returns gcc-6 (on mac) gcc-5 (on linux) if gcc formula offers gcc-6" do
       software_spec.fails_with(:clang)
       allow(Formulary).to receive(:factory).with("gcc").and_return(double(version: "6.0"))
+      if OS.mac?
+        expect(selector.compiler).to eq("gcc-6")
+      else
+        expect(selector.compiler).to eq("gcc-5")
+      end
```

Comment:
```suggestion
      if OS.linux?
        expect(selector.compiler).to eq("gcc-5")
      else
        expect(selector.compiler).to eq("gcc-6")
      end
```
Both macOS and generic return `gcc-6`.

Reply:
Yes, it's a little bit confusing at first, but makes sense.